lib/model/unreads.dart

```diff
@@ -125,6 +125,37 @@ class Unreads extends ChangeNotifier {
 
   int countInDmNarrow(DmNarrow narrow) => dms[narrow]?.length ?? 0;
 
+  int countInNarrow(Narrow narrow) {
+    switch (narrow) {
+      case DmNarrow():
+        return dms[narrow]?.length ?? 0;
```

Comment:
This should reuse `countInDmNarrow`.

Then let's also pull out the other cases as their own methods, along the line I described here:
https://github.com/zulip/zulip-flutter/pull/334#issuecomment-1785873385

Specifically probably like `int countInTopicNarrow(int streamId, String topic)` is most convenient. (The case of DM narrows is unusual in that even when you know you're talking about a DM narrow, the `DmNarrow` value itself is the most convenient way to identify it.)

That way other call sites that know what kind of count they want can be direct about it, just like the recent-DMs view is in #334.